aten/src/ATen/native/PointwiseOps.cpp

```diff
@@ -38,7 +38,8 @@ Tensor& addcmul_out(
     Scalar value) {
   checkBackend("addcmul_cpu", result, self.type().backend());
   auto iter = at::TensorIterator();
-  iter.check_and_add_output(result);
+  iter.set_check_mem_overlap(true);
```

Comment:
I don't see a test for addcmul or addcdiv. It might be better to split this PR in two since it's getting pretty large already:
- first part: Adding the new overlap check and enabling it for TensorIterator::binary_op / TensorIterator::unary_op
- second part: enabling the check for addcmul/ addcdiv / copy_.

Reply:
addressed